File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
def solution(queue1, queue2):
    answer = -1
    total = sum(queue1)+sum(queue2)
    queue3 = queue1+queue2
    q1pointer = 0
    q2pointer = len(queue2)
    sum1 = sum(queue1)
    cnt = 0
    while cnt <= len(queue3)*2:
        if sum1 == total/2:
            return cnt
        elif sum1 < total/2:
            sum1 += queue3[q2pointer%len(queue3)]
            q2pointer+=1
        else:
            sum1 -= queue3[q1pointer%len(queue3)]
            q1pointer+=1
        cnt+=1
    return answer

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    var answer = -1;
    // 3 2 7 2 , 4 6 5 1
    // queue1: 14 0~3
    // queue2: 16 4~7
    
    // 3 , 2 7 2 , 4 6 5 1
    // queue1: 11 1~3
    // queue2: 19 4~8(0)
    
    // 3 , 2 7 2 4, 6 5 1
    // queue1: 15 1~4
    // queue2: 15 5~8
    const queue = [...queue1,...queue2]
    const sum = queue.reduce((prev,cur)=>prev+cur,0)
    
    let sum1 = queue1.reduce((prev,cur)=>prev+cur,0)
    let start1 = 0 // queue1의 시작
    let end1 = queue1.length-1 // queue1의 끝
    let start2 = queue2.length // queue2의 시작
    let end2 = queue.length-1 // queue2의 끝
    
    let cnt = 0
    while(cnt <= queue.length * 2){
        if(sum1<sum/2){
            end1+=1
            sum1 += queue[end1%queue.length]
            cnt+=1
        }else if(sum1 > sum/2){
            sum1 -= queue[start1%queue.length]
            start1+=1
            cnt+=1
        }else{
            answer = cnt
            break
        }
    }
    
    
    
    return answer;
}